File: backend/auth-service/cqrs/handler/command.handler.js
import logger from '../../config/logger.js';
import translation from '../../config/translation.js';

export class CommandHandler {
  constructor() {
    this.handlers = new Map();
  }

  register(commandType, handler) {
    this.handlers.set(commandType, handler);
    logger.info(translation('cqrs.handler.command.logs.registered'), { commandType });
  }

  async dispatch(commandType, command) {
    const handler = this.handlers.get(commandType);
    
    if (!handler) {
      const error = translation('cqrs.handler.command.logs.noHandler', { commandType });
      logger.error(error);
      throw new Error(error);
    }

    try {
      logger.info(translation('cqrs.handler.command.logs.dispatching'), { commandType, command });
      const result = await handler.execute(command);
      logger.info(translation('cqrs.handler.command.logs.success'), { commandType, result });
      return result;
    } catch (error) {
      logger.error(translation('cqrs.handler.command.logs.fail'), { commandType, error, command });
      throw error;
    }
  }
}

export const commandHandler = new CommandHandler(); 

File: backend/ticket-service/cqrs/handler/CommandHandler.py
class CommandHandler:
    def __init__(self):
        self.handlers = {}

    def register(self, command_type, handler):
        self.handlers[command_type] = handler
        print(f"[CommandHandler] Registered handler for: {command_type}")

    async def dispatch(self, command_type, command):
        handler = self.handlers.get(command_type)
        if not handler:
            error = f"No handler registered for command type: {command_type}"
            print(f"[CommandHandler] {error}")
            raise Exception(error)
        try:
            print(f"[CommandHandler] Dispatching command: {command_type} {command}")
            result = await handler.execute(command)
            print(f"[CommandHandler] Success: {command_type} {result}")
            return result
        except Exception as error:
            print(f"[CommandHandler] Fail: {command_type} {error} {command}")
            raise error

    def execute(self, *args, **kwargs):
        raise NotImplementedError("execute metodu override edilmeli.")

# Global instance
command_handler = CommandHandler()
